### PR TITLE
Assert all project source files are monolithic.

### PR DESCRIPTION
**Pull request by SpacePossum**

### CODE REVIEW

**Comment by julienfalque** [comment]
What is the value of this test (real question)?

**Comment by SpacePossum** [comment]
I was close to merge 4953, but than spotted this:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4953/files#diff-24535a6889791dd70f4fe486f14c760fR1280

I would like it as autoreview so I don't miss it again.

**Comment by julienfalque** [comment]
Not sure that would be spotted: this file *is* a monolithic PHP file (no inline HTML or multiple open tag).

**Comment by SpacePossum** [comment]
check, I'll update to make sure there is only one named class declaration in the file!

File: tests/AutoReview/ProjectCodeTest.php [hunk 1]
```diff
@@ -500,6 +502,85 @@ static function ($className) {
...
+                static::assertSame(1, $numberOfClassDefinitions, sprintf('File "%s" contains multiple class definitions.', $file));
+            }
+
+            static::assertSame(0, $numberOfTraitDefinitions, sprintf('File "%s" contains a class definition and a trait definition.', $file));
+        }
+
+        static::assertGreaterThan(0, $numberOfTraitDefinitions + $numberOfClassDefinitions, sprintf('File "%s" does not contains a class, interface or trait definition.', $file));
+    }
+
+    public function provideAllCodeIsMonolithicPhpCases()
```
Comment (kubawerlos) [inline tests/AutoReview/ProjectCodeTest.php:571]: Do we need this provider? Can't we use `provideSrcClassCases` and `provideTestClassCases`?

File: tests/AutoReview/ProjectCodeTest.php [hunk 2]
```diff
@@ -500,6 +502,85 @@ static function ($className) {
         );
     }
 
+    /**
+     * @dataProvider provideAllCodeIsMonolithicPhpCases
+     *
+     * @param string $file
+     */
+    public function testAllCodeIsMonolithicPhp($file)
```
Comment (kubawerlos) [inline tests/AutoReview/ProjectCodeTest.php:510]: I guess the name is to change.

Now it looks quite complicated - isn't the point of the test to see if a file contains only 1 classy element?

Can we just go from the file start and:
 - get next element of kind (class, interface or trait)
 - get next `{`
 - find block end
 - check that there is nothing after it

Wouldn't that be enough? That would work for PR mention as the reason for this test, right?

Comment (SpacePossum) [reply tests/AutoReview/ProjectCodeTest.php:510]: great idea, I'll rewrite soon, thanks!

File: tests/AutoReview/ProjectCodeTest.php [hunk 3]
```diff
@@ -416,6 +413,62 @@ static function (\ReflectionMethod $reflectionMethod) use ($reflectionClass) {
...
+            T_COMMENT,
+            T_DECLARE,
+            T_DOC_COMMENT,
+            T_FINAL,
+            T_LNUMBER,
+            T_NAMESPACE,
+            T_NS_SEPARATOR,
+            T_OPEN_TAG,
+            T_STRING,
+            T_STRING,
```
Comment (kubawerlos) [inline tests/AutoReview/ProjectCodeTest.php:436]: I think we don't need double-check T_STRING 😄

**Comment by julienfalque** [comment]
Wondering: this looks like static analysis concerns. Isn't there some PHPStan extension that implements similar checks out there? It not, wouldn't it be better to implement one?

File: tests/AutoReview/ProjectCodeTest.php (hunk 3 above)
Comment (SpacePossum) [reply tests/AutoReview/ProjectCodeTest.php:436]: : D

**Comment by SpacePossum** [comment]
Maybe, there more of there in the same unit test. I think it would be great a stan plugin but till it exists I think it would be nice ti have here in this form.
When I got time I'll look into making a stan plugin :)

**Pull request merged**